File: Data/initdb.py
from singleton import Database


def initialize_database():
    db = Database()
    cursor = db.get_cursor()

    with open("schema.sql", "r") as f:
        schema = f.read()

    cursor.execute(schema)
    cursor.close()

    print("Database initialized successfully.")


if __name__ == "__main__":
    initialize_database()

File: Application/src/initdb.js
const fs = require('fs');
const path = require('path');
const db = require('../../Data/db'); // Your singleton DB connection

async function initializeDatabase() {
    console.log('Starting database initialization...');

    try {
        // 1. Read the SQL file
        const sqlFilePath = path.join(__dirname, '../../Data/schema.sql');
        const sqlQuery = fs.readFileSync(sqlFilePath, { encoding: 'utf8' });

        // 2. Execute the queries
        await db.query(sqlQuery);
        
        console.log('✅ Database tables initialized successfully!');
    } catch (error) {
        console.error('❌ Error initializing database:', error.message);
    } finally {
        // 3. Exit the process so the script doesn't hang
        process.exit(0);
    }
}

initializeDatabase();